webapp/src/components/common/DownloadRawButton.tsx

```diff
@@ -0,0 +1,85 @@
+import FileUploadIcon from "@mui/icons-material/FileUpload";
+import SplitButton from "./buttons/SplitButton.tsx";
+import { downloadRaw } from "../../services/api/studies/raw";
+import { downloadFile } from "../../utils/fileUtils.ts";
+import { useState } from "react";
+import { StudyMetadata } from "../../common/types.ts";
+import useEnqueueErrorSnackbar from "../../hooks/useEnqueueErrorSnackbar.tsx";
+import { useTranslation } from "react-i18next";
+
+export interface DownloadRawButtonProps {
+  studyId: StudyMetadata["id"];
+  path?: string;
+  disabled?: boolean;
+  label?: string;
+}
+
+const EXPORT_OPTIONS = [
+  { label: "TSV", value: "tsv" },
+  { label: "Excel", value: "xlsx" },
+] as const;
+
+type ExportFormat = (typeof EXPORT_OPTIONS)[number]["value"];
+
+function DownloadRawButton(props: DownloadRawButtonProps) {
+  const { t } = useTranslation();
+  const { studyId, path, disabled, label = t("global.export") } = props;
+  const [isDownloading, setIsDownloading] = useState(false);
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleDownload = async (format: ExportFormat) => {
+    if (!path) {
+      return;
+    }
+
+    setIsDownloading(true);
+
+    const isExcel = format === "xlsx";
+
+    try {
+      const res = await downloadRaw({
+        studyId,
+        path,
+        format,
+        header: isExcel,
+        index: isExcel,
+      });
+
+      downloadFile(
+        res,
+        `matrix_${studyId}_${path.replace("/", "_")}.${format}`,
+      );
+    } catch (err) {
+      enqueueErrorSnackbar(t("global.download.error"), String(err));
+    }
+
+    setIsDownloading(false);
+  };
```

Comment:
Why? The raw API in not specific for the download of matrices. This is why I named the component with 'Raw'. When you want to download a button that use this API, you just use it, instead of repeat the same logic every times.